packages/model-viewer/src/test/model-viewer-spec.ts

```diff
@@ -0,0 +1,169 @@
+import {$renderer} from '../model-viewer-base.js';
+import {ModelViewerElement} from '../model-viewer.js';
+import {Constructor} from '../utilities.js';
+
+import {assetPath, waitForEvent} from './helpers.js';
+import {BasicSpecTemplate} from './templates.js';
+
+const expect = chai.expect;
+
+const LIGHTROOM_PATH = 'environments/lightroom_14b.hdr';
+const SUNRISE_HDR_PATH = 'environments/spruit_sunrise_1k_HDR.hdr';
+const SUNRISE_LDR_PATH = 'environments/spruit_sunrise_1k_LDR.jpg';
+
+const COMPONENTS_PER_PIXEL = 4;
+
+const setupModelViewer =
+    async (modelViewer: ModelViewerElement, lighting: string) => {
+  modelViewer.style.width = '100px';
+  modelViewer.style.height = '100px';
+
+  modelViewer.style.backgroundColor = 'rgba(255,255,255,0)';
+
+  modelViewer.src = assetPath(
+      'models/glTF-Sample-Models/2.0/MetalRoughSpheres/glTF/MetalRoughSpheres.gltf');
+
+  const lightingPath = assetPath(lighting);
+  modelViewer.environmentImage = lightingPath;
+  modelViewer.skyboxImage = lightingPath;
+
+  modelViewer.minCameraOrbit = 'auto auto 12m';
+  modelViewer.maxCameraOrbit = 'auto auto 12m';
+  modelViewer.cameraOrbit = '0deg 90deg 12m';
+  modelViewer.cameraTarget = '0m 0m 0m';
+  modelViewer.fieldOfView = '45deg';
+
+  await waitForEvent(modelViewer, 'poster-dismissed');
+};
+
+// TODO(sun765): this only test whether the screenshot
+// is colorless or not. Replace this with more robust
+// test in later pr.
+function testFidelity(screenshotContext: WebGLRenderingContext) {
+  const width = screenshotContext.drawingBufferWidth;
+  const height = screenshotContext.drawingBufferHeight;
+
+  const pixels = new Uint8Array(width * height * COMPONENTS_PER_PIXEL);
+  // this function reads in the bottom-up direction from the coordinate
+  // specified ((0,0) is the bottom-left corner).
+  screenshotContext.readPixels(
```

Comment:
Although im not using alpha value at all, i think it may be useful for later work, so i read the pixels in rgba format.